storage/local/chunk/chunk.go

```diff
@@ -402,23 +409,23 @@ func newIndexAccessingChunkIterator(len int, acc indexAccessor) *indexAccessingC
 	return &indexAccessingChunkIterator{
 		len:       len,
 		pos:       -1,
-		lastValue: ZeroSamplePair,
+		lastValue: model.SamplePair{Timestamp: model.Earliest},
```

Comment:
This might be the right time to move `ZeroSamplePair` into `model`. The problem is that the "natural" zero value is a reasonable data point that could come from a real scrape. Hardcoding the `model.Earliest` sentinel value at various places makes things worse.

While it is fine to see this change as an iterative approach towards proper separation (and thus allow some bad smell), me should not make anything worse than before.